### ISSUE TITLE
Implement map-pointer, de-hardcode pointer bindings

### ISSUE THREAD

**Issue by ifreund** [ifreund/river#64]
For maximum flexibility we should allow running arbitrary commands on mouse button click/release. For consistency, we should also create commands for moving/resizing views with the mouse. This would look something like
```
riverctl map normal Mod4 BTN_LEFT enter-move
riverctl map normal Mod4 BTN_LEFT --release exit-move
```
I'm not 100% what the right interface would be for specifying release/press. One possibility is the optional flag used in the above example. Another option would be making press/release required like so:
```
riverctl map normal Mod4 press BTN_LEFT enter-move
riverctl map normal Mod4 release BTN_LEFT enter-move
```
This is however quite verbose.

This is certainly more complex than having mouse actions hardcoded, but I believe that it is fitting for river as it provides maximum flexibility and gives power to the user. Having mappings which run on key release is likely useful for other things as well (e.g. something like #26).

**Comment by Leon-Plickat** [comment]
Hmmm... I think the `--release` bindings would *massively* complicate button handling. As a reference: When entering move mode, we have to check whether the pointer modifier is pressed, so that views are not moved when a user just wants to click around in a window. However, when *leaving* moving mode, we explicitly do not care whether the pointer modifier is pressed, so that a user can stop moving a window when they release the pointer modifier first and the button second.

There are two possible solutions to approach this.

1. If we *really* want `--release` bindings, we'd have to implement the possibility of having multiple pointer modifiers, an individual one for each button bind. This is pretty tedious, because modifiers are stored per keyboard, not per seat and a seat can have multiple keyboards, meaning it would require some looping (this is the reason I simply added a simple boolean `pointer_modifier` to seat when I implemented pointer move). Also this is non-trivial, so prepare for at least three or four "mouse bindings do not work as expected" false bug reports per month. The added complexity is also not really worth the gain of functionality.

2. We will not implement `--release` and keep a single pointer modifier. When binding an action to a mouse button (like entering move mode), the corresponding action (in this example leaving move mode), if it exists (closing a view obviously only has no follow-up action), will be implicitly bound to the same key on release. This would be my preferred because it is considerably easier to implement (in `Cursor.zig` we'd probably only have to change the one single switch that hardcode-assigns actions to buttons  to something more modular) and because this would result in expectable behaviour.

**Comment by ifreund** [comment]
I think you're right, `--release` and `enter-move`/`exit-move` commands are a bad idea and aren't a good complexity to usefulness tradeoff.

As an alternative, I propose adding a new `map-pointer` command which takes input codes such as `BTN_LEFT` instead of keysyms. In addition to running any normal command, `map-pointer` would accept two special `move` and `resize` commands.

Thus, the current behavior of moving with left click and closing a view with middle click could be achieved with:
```
riverctl map-pointer normal Mod4 BTN_LEFT move
riverctl map-pointer normal Mod4 BTN_MIDDLE close
```

Implementation wise, this would probably mean getting rid of the `pointer_modifier` bool in `Seat` and instead querying the Keyboard's modifier state directly when handling a mouse button press, matching against the current pointer bindings.

**Comment by ifreund** [comment]
Implemented this for move and resize in 989e7aaeda44b02bdb92455dfe7b58dafc28435c
